packages/connect-migrate/src/lib/migrate-dependencies.ts

```diff
@@ -0,0 +1,156 @@
+// Copyright 2021-2024 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { PackageJson } from "./package-json";
+import {
+  intersects as semverIntersects,
+  satisfies as semverSatisfies,
+  valid as semverValid,
+  validRange as semverValidRange,
+} from "semver";
+
+/**
+ * Migrate matching dependencies, removing, upgrading, or renaming them.
+ */
+export type DependencyMigration = DependencyReplacement | DependencyRemoval;
+/**
+ * Replace all dependencies matching "from" with "to".
+ */
+type DependencyReplacement = {
+  from: { name: string; range: string };
+  to: { name?: string; version: string };
+};
+/**
+ * Remove all dependencies matching name and version range.
+ */
+type DependencyRemoval = {
+  remove: { name: string; range: string };
+};
```

Comment:
Instead of only DependencyReplacement, we're supporting this type as well. For the existing migration to v0.13.1, nothing changes except symbol names.